src/main/java/com/bio4j/model/uniprot/programs/ImportProteinInteractions.java

```diff
@@ -122,78 +123,106 @@ protected void importProteinInteractions(File inFile, File dbFolder) {
 
         Logger.getLogger(ImportProteinInteractions.class.getName()).log(Level.SEVERE, null, ex);
       }
-
     }
   }
 
-  private void importProteinInteractionsWithSource(
-    XMLElement entryXMLElem,
-    UniProtGraph<I,RV,RVT,RE,RET> graph,
-    Protein<I,RV,RVT,RE,RET> protein
-  )
-  {
-
-    List<Element> comments = entryXMLElem.asJDomElement().getChildren(COMMENT_TAG_NAME);
+  /*
+    By direct inspection of UniProt XML file I have concluded that:
 
-    for (Element commentElem: comments) {
+    1. the source id is always a protein, I don't know why
+    2. target can be either an isoform or a protein, and the id is in `interactant/id`. Isoform ids *look* to be `${protein.id}-{number}`
+    3. the `intactId`s have nothing to do with UniProt, and should be dropped
 
-      String commentTypeSt = commentElem.getAttributeValue(COMMENT_TYPE_ATTRIBUTE);
-
-      //----------interaction----------------
-      if (commentTypeSt.equals(COMMENT_TYPE_INTERACTION)) {
-
-        List<Element> interactants = commentElem.getChildren("interactant");
-        Element interactant1 = interactants.get(0);
-        Element interactant2 = interactants.get(1);
-        Element organismsDiffer = commentElem.getChild("organismsDiffer");
-        Element experiments = commentElem.getChild("experiments");
-        String intactId1St = interactant1.getAttributeValue("intactId");
-        String intactId2St = interactant2.getAttributeValue("intactId");
-        String organismsDifferSt = "";
-        String experimentsSt = "";
-        if (intactId1St == null) {
-          intactId1St = "";
-        }
-        if (intactId2St == null) {
-          intactId2St = "";
-        }
-        if (organismsDiffer != null) {
-          organismsDifferSt = organismsDiffer.getText();
-        }
-        if (experiments != null) {
-          experimentsSt = experiments.getText();
-        }
+    Note that because of 1. it would be easier if we passed the protein as an argument.
 
-        //----now we try to retrieve the interactant 2 accession--
-        String interactant2AccessionSt = interactant2.getChildText("id");
-        long protein2Id = -1;
-        if (interactant2AccessionSt != null) {
+    #### Remark
 
-          Optional<Protein<I,RV,RVT,RE,RET>> protein2Optional = graph.proteinAccessionIndex().getVertex(interactant2AccessionSt);
-
-          if(!protein2Optional.isPresent()) {
-
-            Optional<Isoform<I,RV,RVT,RE,RET>> isoformOptional = graph.isoformIdIndex().getVertex(interactant2AccessionSt);
-
-            if(isoformOptional.isPresent()) {
+    Again based on manual inspection it looks like protein-protein interactions are **not** *duplicated* in the UniProt XML file. I do not know if this implies a semantic direction or not. So, if you want to get all with which a given protein interacts, you need to get both inV and outV of protein-protein interaction edges.
```

Comment:
I think this remark should be taken out on discussion with @rtobes, otherwise it'll get lost in the code..